File: java/arcade/AvoidObstacles.java
import java.util.Arrays;

public final class AvoidObstacles {

	int avoidObstacles(int[] inputArray) {
		boolean b = true;
	    for(int i = 2; true; i++) {
	        for(int f : inputArray) {
	        	if(f % i == 0) {
	        		b = !b;
	        		break;
	        	}
	        }
	        if(b) { return i; }
	        b = true;
	    }
    
	}
}

File: Arcade/Intro/Ruby/avoidObstacles.rb
def avoidObstacles(inputArray)
	k, i = true, 2
    loop do
        inputArray.each do |e|
            if e % i == 0
                k = !k
                break
            end
        end
        return i if k
        k, i = true, i + 1
    end
end